### PR TITLE
feat: refactor number type

### PR DESCRIPTION
**Pull request by fjchen7**
This PR is mainly on type refactor, to refactor all fields of Uint32/Unit64 in [blockchain.mol](https://github.com/nervosnetwork/ckb/blob/develop/util/types/schemas/blockchain.mol) and [CKB RPC doc](https://github.com/nervosnetwork/ckb/tree/develop/rpc) into int/long in Java SDK.

Some details on code implementations
- To focus on type refactor, I didn't check business logic carefully (most in ckb indexer and some utils methods) and only change code where the refactored field is used. This work to check business logic should be put into other PR.
- Also, types used only by mercury will be refactor in the future, not covered in this PR.
- ~~I didn't use unsigned method to manipulate capacity as they won't exceed LONG.MAX_VALUE in real cases and can work correctly with signed operations.~~

### CODE REVIEW

File: ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/common/Record.java [hunk 1]
```diff
@@ -15,9 +15,9 @@ public class Record {
   @SerializedName("ownership")
   public Ownership ownership;
 
-  public BigInteger amount;
+  public long amount;
```
Comment (TheWaWaR) [inline ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/common/Record.java:18]: amount can be udt amount, thus the type is `u128`, so here use `long` as data type is not enough.

File: ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/req/ToInfo.java [hunk 2]
```diff
@@ -4,9 +4,9 @@
 
 public class ToInfo {
   public String address;
-  public BigInteger amount;
+  public long amount;
```
Comment (TheWaWaR) [inline ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/req/ToInfo.java:7]: Also can be UDT amount here.

File: ckb/src/main/java/org/nervos/ckb/transaction/CellCollector.java [hunk 3]
```diff
@@ -58,41 +58,42 @@ public CollectResult collectInputs(
...
       cellInputList.add(cellInput);
       cellInputs.add(cellInput);
       witnesses.add("0x");
       transaction.inputs = cellInputs;
       transaction.witnesses = witnesses;
-      BigInteger sumNeedCapacity =
+      long sumNeedCapacity =
           calSumNeedCapacity(feeRate, transaction, changeOutput, haveChangeOutput, needCapacity);
-      if (inputsCapacity.compareTo(sumNeedCapacity) > 0) {
+      if (inputsCapacity > sumNeedCapacity) {
```
Comment (TheWaWaR) [inline ckb/src/main/java/org/nervos/ckb/transaction/CellCollector.java:96]: Please use [Long.compareUnsigned](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Long.html#compareUnsigned(long,long))

File: core/src/main/java/org/nervos/ckb/service/adapter/IntegerTypeAdapter.java [hunk 4]
```diff
@@ -25,11 +18,6 @@ public Integer deserialize(JsonElement json, Type typeOfT, JsonDeserializationCo
       return json.getAsInt();
     }
     String hexValue = json.getAsString().substring(2);
-    // deserialize 0xffffffff to -1 for outpoint index in coinbase transaction
-    if (Objects.equals("ffffffff", hexValue)) {
-      return -1;
-    } else {
-      return Integer.valueOf(hexValue, 16);
-    }
+    return Integer.parseUnsignedInt(hexValue, 16);
```
Comment (TheWaWaR) [inline core/src/main/java/org/nervos/ckb/service/adapter/IntegerTypeAdapter.java:21]: Use [parseUnsignedLong](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Long.html#parseUnsignedLong(java.lang.CharSequence,int,int,int)) instead?

File: ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/req/ToInfo.java (hunk 2 above)
Comment (fjchen7) [reply ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/req/ToInfo.java:7]: So far mercury change is not ready for test. I plan to refactor this part after mercury change so just leave them here now.

File: core/src/main/java/org/nervos/ckb/service/adapter/IntegerTypeAdapter.java (hunk 4 above)
Comment (fjchen7) [reply core/src/main/java/org/nervos/ckb/service/adapter/IntegerTypeAdapter.java:21]: No. This method returns a Integer.

File: ckb/src/main/java/org/nervos/ckb/transaction/CellCollector.java (hunk 3 above)
Comment (fjchen7) [reply ckb/src/main/java/org/nervos/ckb/transaction/CellCollector.java:96]: I mention this in the PR description

> I didn't use unsigned method to manipulate capacity as they won't exceed LONG.MAX_VALUE in real cases and can work correctly with signed operations.

File: ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/common/Record.java (hunk 1 above)
Comment (fjchen7) [reply ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/common/Record.java:18]: You are right. Here should be changed

File: ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/req/ToInfo.java (hunk 2 above)
Comment (TheWaWaR) [reply ckb-mercury-sdk/src/main/java/org/nervos/mercury/model/req/ToInfo.java:7]: ok

**Review by quake — APPROVED** [review]

**Pull request merged**